08_deploy_bmo.sh

```diff
@@ -12,10 +12,36 @@ export BMOPATH="$GOPATH/src/github.com/metal3-io/baremetal-operator"
 # Make a local copy of the baremetal-operator code to make changes
 cp -r $BMOPATH/deploy ocp/.
 sed -i 's/namespace: .*/namespace: openshift-machine-api/g' ocp/deploy/role_binding.yaml
+cp $SCRIPTDIR/operator_ironic.yaml ocp/deploy
+cp $SCRIPTDIR/ironic_bmo_configmap.yaml ocp/deploy
+sed -i "s#__RHCOS_IMAGE_URL__#${RHCOS_IMAGE_URL}#" ocp/deploy/ironic_bmo_configmap.yaml
+
+# Kill the dnsmasq container on the host since it is performing DHCP and doesn't
+# allow our pod in openshift to take over.
+for name in dnsmasq ironic-inspector ; do
+    sudo podman ps | grep -w "$name$" && sudo podman stop $name
+done
 
 # Start deploying on the new cluster
 oc --config ocp/auth/kubeconfig apply -f ocp/deploy/service_account.yaml --namespace=openshift-machine-api
 oc --config ocp/auth/kubeconfig apply -f ocp/deploy/role.yaml --namespace=openshift-machine-api
 oc --config ocp/auth/kubeconfig apply -f ocp/deploy/role_binding.yaml
 oc --config ocp/auth/kubeconfig apply -f ocp/deploy/crds/metal3_v1alpha1_baremetalhost_crd.yaml
-oc --config ocp/auth/kubeconfig apply -f ocp/deploy/operator.yaml --namespace=openshift-machine-api
+
+oc --config ocp/auth/kubeconfig apply -f ocp/deploy/ironic_bmo_configmap.yaml --namespace=openshift-machine-api
+# I'm leaving this as is for debugging but we could easily generate a random password here.
+oc --config ocp/auth/kubeconfig delete secret mariadb-password --namespace=openshift-machine-api || true
+oc --config ocp/auth/kubeconfig create secret generic mariadb-password --from-literal password=password --namespace=openshift-machine-api
+
+oc --config ocp/auth/kubeconfig adm --as system:admin policy add-scc-to-user privileged system:serviceaccount:openshift-machine-api:baremetal-operator
+oc --config ocp/auth/kubeconfig apply -f ocp/deploy/operator_ironic.yaml -n openshift-machine-api
+
+# Sadly I don't see a way to get this from the json..
+POD_NAME=$(oc --config ocp/auth/kubeconfig get pods -n openshift-machine-api | grep metal3-baremetal-operator | cut -f 1 -d ' ')
+
+# Make sure our pod is running.
+echo "Waiting for baremetal-operator pod to become ready"
```

Comment:
Why do we need to wait?  It shouldn't be required to block here

Reply:
Yeah it's probably optional.  I haven't tested the whole thing without the wait yet though.  I'll give it a go.  On the other hand we could use this to catch errors early.

Reply:
Fair enough.

Reply:
FWIW in my testing this was helpful when one of the containers went into CrashLoopBackoff, it meant I could start investigating when it became clear the pod was wedged and not starting correctly.

I don't have a strong opinion but for development having some verbose monitoring of the pod startup is probably no bad thing?

Reply:
yeah I'm 50/50 on it.  I do kinda like how it can catch an issue with the pod, but for CI it's probably not useful.